src/frontend/src/lib/stores/convert.store.ts

```diff
@@ -0,0 +1,75 @@
+import { exchanges } from '$lib/derived/exchange.derived';
+import { balancesStore } from '$lib/stores/balances.store';
+import type { Token } from '$lib/types/token';
+import { nonNullish } from '@dfinity/utils';
+import { BigNumber } from '@ethersproject/bignumber';
+import { derived, writable, type Readable } from 'svelte/store';
+
+export interface ConvertData {
+	sourceToken: Token;
+	destinationToken: Token;
+}
+
+export interface ConvertStore extends Readable<ConvertData> {
+	set: (data: ConvertData) => void;
+}
+
+export const initConvertStore = (data: ConvertData): ConvertStore => {
+	const { subscribe, set: setStore } = writable<ConvertData>(data);
+
+	return {
+		subscribe,
+
+		set(data: ConvertData) {
+			setStore(data);
+		}
+	};
+};
+
+export const initConvertContext = (convertData: ConvertData): ConvertContext => {
+	const convertStore = initConvertStore(convertData);
+
+	const sourceToken = derived([convertStore], ([{ sourceToken }]) => sourceToken);
+	const destinationToken = derived([convertStore], ([{ destinationToken }]) => destinationToken);
+
+	const sourceTokenBalance = derived(
+		[balancesStore, sourceToken],
+		([$balancesStore, $sourceToken]) =>
```

Comment:
That's a new pattern. I like the idea, but I'm not sure if I like placing it in the `/src/**/stores` package. It gives me a bit of a recursive import vibe, like `derived` using `stores`, which then uses `derived`. What do you think?

Reply:
I think `send.store` uses a similar pattern for `sendBalance`. Maybe in another PR we can introduce a new folder `contexts` (or something like `context-stores`) and move there all similar context files where no custom stores needed? It does not solve the "recursion" feeling completely, though.

Reply:
Ah yeah right, don't know why I forgot about it. I think at some point we already discussed the idea of a `contexts` package with @AntonioVentilii so sounds good to me.